packages/flutter/test/widgets/selectable_text_test.dart

```diff
@@ -1475,6 +1475,59 @@ void main() {
     expect(topLeft.dx, equals(399.0));
   });
 
+  testWidgets('Tapping outside SelectableText clears the selection', (WidgetTester tester) async {
+    Future<void> setAppLifecycleState(AppLifecycleState state) async {
+      final ByteData? message = const StringCodec().encodeMessage(state.toString());
+      await TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger
+          .handlePlatformMessage('flutter/lifecycle', message, (_) {});
+    }
+    await tester.pumpWidget(
+      const MaterialApp(
+        home: Material(
+          child: Center(
+            child: Column(
+              children: <Widget>[
+                SelectableText('first selectable text'),
+                SelectableText('second selectable text'),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+    // Setting the app lifecycle state to AppLifecycleState.resumed to simulate
+    // an applications default running mode, i.e. the application window is focused.
+    await setAppLifecycleState(AppLifecycleState.resumed);
+    await tester.pumpAndSettle();
+
+    // First tap on the first SelectableText sets the cursor.
+    await tester.tapAt(textOffsetToPosition(tester, 5));
+    await tester.pumpAndSettle();
+
+    final EditableText editableTextWidgetFirst = tester.widget(find.byType(EditableText).first);
+    final TextEditingController controllerA = editableTextWidgetFirst.controller;
+    final EditableText editableTextWidgetSecond = tester.widget(find.byType(EditableText).last);
+    final TextEditingController controllerB = editableTextWidgetSecond.controller;
+
+    expect(controllerA.selection, const TextSelection.collapsed(offset: 5));
+    expect(controllerB.selection, TextRange.empty);
+
+    // Tapping on the second SelectableText sets the cursor on it, and clears the selection from
+    // the first SelectableText.
+    final Offset selectableTextStart = tester.getTopLeft(find.byType(SelectableText).last);
+    await tester.tapAt(selectableTextStart);
+    await tester.pumpAndSettle();
+    expect(controllerA.selection, TextRange.empty);
+    expect(controllerB.selection, const TextSelection.collapsed(offset: 0));
+
+    // Setting the app lifecycle state to AppLifecycleState.inactive to simulate
+    // a lose of window focus. Selection should remain the same.
+    await setAppLifecycleState(AppLifecycleState.inactive);
+    await tester.pumpAndSettle();
+    expect(controllerA.selection, TextRange.empty);
+    expect(controllerB.selection, const TextSelection.collapsed(offset: 0));
+  }, variant: TargetPlatformVariant.all());
```

Comment:
Is it worth adding `.all` here instead of just omitting it? You do have a `switch` on the platform above, but maybe I think it doesn't affect this particular test.

Reply:
I'm okay with either, but I usually lean towards `.all` if possible, and not platform specific, for more test coverage. What do you think?